src/Sylius/Component/Core/Model/Product.php

```diff
@@ -330,24 +336,58 @@ public function getPrice()
     /**
      * {@inheritdoc}
      */
-    public function getImage()
+    public function getImages()
     {
-        if (null === $this->getFirstVariant()) {
-            return null;
+        return $this->images;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getImageByCode($code)
```

Comment:
Since these methods are common to any entity containing an image i think it makes sense to introduce `ImageOwnerInterface` for DX sake.